api/src/main/java/org/pmiops/workbench/api/ClusterController.java

```diff
@@ -186,115 +183,97 @@
-        body.getWorkspaceNamespace(), body.getWorkspaceId(), WorkspaceAccessLevel.READER);
-    workspaceService.validateActiveBilling(body.getWorkspaceNamespace(), body.getWorkspaceId());
+        dbWorkspace.getWorkspaceNamespace(),
+        dbWorkspace.getFirecloudName(),
+        WorkspaceAccessLevel.WRITER);
+    workspaceService.validateActiveBilling(
+        dbWorkspace.getWorkspaceNamespace(), dbWorkspace.getFirecloudName());
 
-    FirecloudWorkspace fcWorkspace;
+    FirecloudWorkspace firecloudWorkspace;
     try {
-      fcWorkspace =
+      firecloudWorkspace =
           fireCloudService
-              .getWorkspace(body.getWorkspaceNamespace(), body.getWorkspaceId())
+              .getWorkspace(dbWorkspace.getWorkspaceNamespace(), dbWorkspace.getFirecloudName())
               .getWorkspace();
     } catch (NotFoundException e) {
       throw new NotFoundException(
           String.format(
               "workspace %s/%s not found or not accessible",
-              body.getWorkspaceNamespace(), body.getWorkspaceId()));
+              dbWorkspace.getWorkspaceNamespace(), dbWorkspace.getFirecloudName()));
     }
-    DbCdrVersion cdrVersion =
-        workspaceService
-            .getRequired(body.getWorkspaceNamespace(), body.getWorkspaceId())
-            .getCdrVersion();
+    DbCdrVersion cdrVersion = dbWorkspace.getCdrVersion();
 
     // For the common case where the notebook cluster matches the workspace
     // namespace, simply name the directory as the workspace ID; else we
     // include the namespace in the directory name to avoid possible conflicts
     // in workspace IDs.
-    String gcsNotebooksDir = "gs://" + fcWorkspace.getBucketName() + "/notebooks";
-    Timestamp now = new Timestamp(clock.instant().toEpochMilli());
-    long workspaceId =
-        workspaceService
-            .getRequired(body.getWorkspaceNamespace(), body.getWorkspaceId())
-            .getWorkspaceId();
+    String gcsNotebooksDir = "gs://" + firecloudWorkspace.getBucketName() + "/notebooks";
+    long workspaceId = dbWorkspace.getWorkspaceId();
 
     body.getNotebookNames()
         .forEach(
             notebook ->
                 userRecentResourceService.updateNotebookEntry(
                     workspaceId, userProvider.get().getUserId(), gcsNotebooksDir + "/" + notebook));
-    String workspacePath = body.getWorkspaceId();
-    if (!projectName.equals(body.getWorkspaceNamespace())) {
-      workspacePath =
-          body.getWorkspaceNamespace()
-              + FireCloudService.WORKSPACE_DELIMITER
-              + body.getWorkspaceId();
-    }
 
+    String workspacePath = dbWorkspace.getFirecloudName();
     String editDir = "workspaces/" + workspacePath;
     String playgroundDir = "workspaces_playground/" + workspacePath;
     String targetDir = body.getPlaygroundMode() ? playgroundDir : editDir;
 
     leonardoNotebooksClient.createStorageLink(
-        projectName,
-        clusterName,
+        workspaceNamespace,
```

Comment:
Can we rename `targetDir` to `notebookLocalDir`? Also, collecting lines 269-272 into a helper method would be nice.